doc/examples/edges/plot_skeleton.py

```diff
@@ -6,59 +6,140 @@
 border pixels are identified and removed on the condition that they do not
 break the connectivity of the corresponding object.
-
-This module provides an example of calling the routine and displaying the
-results. The input is a 2D ndarray, with either boolean or integer elements.
-In the case of boolean, 'True' indicates foreground, and for integer arrays,
-the foreground is 1's.
 """
 from skimage.morphology import skeletonize
-from skimage import draw
-import numpy as np
+from skimage import data
 import matplotlib.pyplot as plt
+from skimage.util import invert
 
-# an empty image
-image = np.zeros((400, 400))
-
-# foreground object 1
-image[10:-10, 10:100] = 1
-image[-100:-10, 10:-10] = 1
-image[10:-10, -100:-10] = 1
-
-# foreground object 2
-rs, cs = draw.line(250, 150, 10, 280)
-for i in range(10):
-    image[rs + i, cs] = 1
-rs, cs = draw.line(10, 150, 250, 280)
-for i in range(20):
-    image[rs + i, cs] = 1
-
-# foreground object 3
-ir, ic = np.indices(image.shape)
-circle1 = (ic - 135)**2 + (ir - 150)**2 < 30**2
-circle2 = (ic - 135)**2 + (ir - 150)**2 < 20**2
-image[circle1] = 1
-image[circle2] = 0
+# Invert the horse image
+image = invert(data.horse())
 
 # perform skeletonization
 skeleton = skeletonize(image)
 
 # display results
-fig, (ax1, ax2) = plt.subplots(nrows=1, ncols=2, figsize=(8, 4.5),
-                               sharex=True, sharey=True,
-                               subplot_kw={'adjustable': 'box-forced'})
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(8, 4),
+                         sharex=True, sharey=True,
+                         subplot_kw={'adjustable': 'box-forced'})
+
+ax = axes.ravel()
+
+ax[0].imshow(image, cmap=plt.cm.gray)
+ax[0].axis('off')
+ax[0].set_title('original', fontsize=20)
+
+ax[1].imshow(skeleton, cmap=plt.cm.gray)
+ax[1].axis('off')
+ax[1].set_title('skeleton', fontsize=20)
+
+fig.tight_layout()
+plt.show()
+
+######################################################################
+# **skeletonize vs skeletonize 3d**
+#
+# ``skeletonize`` [Zha84]_ works by making successive passes of
+# the image, removing pixels on object borders. This continues until no
+# more pixels can be removed.  The image is correlated with a
+# mask that assigns each pixel a number in the range [0...255]
+# corresponding to each possible pattern of its 8 neighbouring
+# pixels. A look up table is then used to assign the pixels a
+# value of 0, 1, 2 or 3, which are selectively removed during
+# the iterations.
+#
+# ``skeletonize_3d`` [Lee94]_ uses an octree data structure to examine a 3x3x3
+# neighborhood of a pixel. The algorithm proceeds by iteratively sweeping
+# over the image, and removing pixels at each iteration until the image
+# stops changing. Each iteration consists of two steps: first, a list of
+# candidates for removal is assembled; then pixels from this list are
+# rechecked sequentially, to better preserve connectivity of the image.
+
```

Comment:
Could you please say that ``skeletonize_3d`` is meant to be used mostly on 3-D images, but that here you show the result on a 2-D image (or something along these lines)? After this, or if you convince me that it's not necessary, I'll merge because this PR already received several :+1: and is long due!